### PR TITLE
sort buffer list by display time in persp-ivy-switch-buffer

### PR DESCRIPTION
**Pull request by QiangF**
fix https://github.com/nex3/perspective-el/issues/142

### CODE REVIEW

File: perspective.el [hunk 1]
```diff
@@ -30,6 +30,7 @@
 ;;; Code:
 
 (require 'cl-lib)
+(require 'dash)
```
Comment (gcv) [inline perspective.el:33]: I'm trying to keep Perspective dependency-free. Even dash is too much here, and its functionality should be reimplemented.

File: perspective.el [hunk 2]
```diff
@@ -1259,7 +1260,8 @@ PERSP-SET-IDO-BUFFERS)."
              (append
               (list
                (format "Switch to buffer (%s): " (persp-current-name))
-               (cl-remove-if #'null (mapcar #'buffer-name (persp-current-buffers)))
+               (cl-remove-if #'null (mapcar #'buffer-name (-intersection (buffer-list)
```
Comment (gcv) [inline perspective.el:1263]: I'm really confused how `-intersection` helps sort by access time here. Please clarify.

**Review by gcv — CHANGES_REQUESTED** [review]

File: perspective.el (hunk 2 above)
Comment (QiangF) [reply perspective.el:1263]: By default (buffer-list) is sorted by access time and the order is kept in the -intersection operation.

File: perspective.el (hunk 1 above)
Comment (QiangF) [reply perspective.el:33]: I think we can use seq-intersection, or why not keep persp-current-buffers sorted by time in the first place.

Comment (gcv) [reply perspective.el:33]: Please switch to `seq-intersection` and verify that it still works. If it breaks CI for older version of Emacs, we can maybe hack around the problem.

File: perspective.el (hunk 2 above)
Comment (gcv) [reply perspective.el:1263]: Very clever! Please add a code comment with this explanation.

**Comment by gcv** [comment]
Thanks!

**Pull request merged**